### DISCUSSION TITLE
How should one model optical splitters

### DISCUSSION THREAD

**Discussion by apanagio** [nautobot/nautobot#3709] [Q&A]
Do you have any suggestions on how to model an optical splitter?
A 1:2 splitter is a device that connects 2 cables with a third one, in a way that whatever comes in in any of the 2 cables gets transmitted to the third one and whatever comes in to the third cable gets transmitted to both other cables.
Respectively there are more complex splitters (1:4, 1:32, 2:32, 2:64, ...)  
https://community.fs.com/blog/what-is-a-fiber-optic-splitter-2.html

The way I follow for modelling 1:n splitter is to declare a device with 1 rear port with n positions and n front ports. Then I connect each front port with a position of the rear.
Do you see any flaws in that method?
Is there a better way to do it?

For modeling a 2:4 splitter this method cannot be followed since it would require front ports with 2 positions and they don't exist in nautobot yet. What I think one should do is create two devices 1:2 and 1:4 and then connect with a cable the two rear ports.
Maybe a virtual chassis could be used to help managing the devices.
Does it sound reasonable?
Do you think front ports with multiple positions would be considered a useful addition in nautobot?

**Comment by bryanculver** [comment]
Hi @apanagio!

We have an open issue to address cable splitters/breakout cables: https://github.com/nautobot/nautobot/issues/1743

In the interim, you should be able to use relationships or custom fields to denote which rear port is associated with which front port and when or by following a number or naming convention.
